core/src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -945,7 +960,8 @@ private void evaluateTriggerable(Triggerable t, TreeReference anchorRef) {
      * when we support dependent XPath Steps (IE: /path/to//)
      */
     public void addChildrenOfReference(TreeReference original, Vector<TreeReference> toAdd) {
-        for (TreeReference ref : exprEvalContext.expandReference(original)) {
+        Vector<TreeReference> refs = exprEvalContext.expandReference(original);
```

Comment:
Question: Are these references genericized when this gets fired? IE: Are the treerefrences

/data/path/to/node

or 

/data[1]/path[1]/to[3]/node[1]